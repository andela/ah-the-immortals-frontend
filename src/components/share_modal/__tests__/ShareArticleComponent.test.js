import { mount } from 'enzyme';
import React from 'react';
import ShareModal from '../ShareArticleComponent';


describe('Test the successful mounting of the share article component', () => {
  const props = {
    article: {
      slug: 'test-stlug'
    }
  };
  const wrapper = mount(
    <ShareModal {...props} />
  );
  it('Test successful mount', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Simulates clicking of show button',()=>{
    const showButton = wrapper.find('button').first();
    showButton.simulate('click');
    expect(showButton.exists()).toEqual(true);
  });
  it('Simulates clicking of close button',()=>{
    const showButton = wrapper.find('button').first();
    showButton.simulate('click');
    const closeButton = wrapper.find('button[id="close-button"]');
    closeButton.simulate('click');
    expect(closeButton.exists()).toEqual(true);
  });
});
