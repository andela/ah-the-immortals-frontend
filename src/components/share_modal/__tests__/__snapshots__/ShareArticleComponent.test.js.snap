// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the successful mounting of the share article component Test successful mount 1`] = `ReactWrapper {}`;
